rust/template/ddlog.h

```diff
@@ -125,14 +126,53 @@ extern ddlog_prog ddlog_run(unsigned int workers,
  *
  * This is a debugging feature used to record DDlog commands issued through
  * functions in this file in a DDlog command file that can later be replayed
- * throught the CLI interface.
+ * through the CLI interface.
  *
  * `fd` - file descriptor.  Passing -1 in this argument instructs DDlog to stop
  * recording. The caller is responsible for opening and closing the file.  They
  * can inject additional commands, e.g., `echo` to the file.
+ *
+ * IMPORTANT: this function is _not_ thread-safe and must not be invoked
+ * concurrently with other functions in this API.
  */
 extern int ddlog_record_commands(ddlog_prog hprog, int fd);
 
+/*
```

Comment:
I find it a bit odd that the most extensive documentation is on the C API.
But I don't have an alternative proposal of where to locate it.
Perhaps we can make a table somewhere in an MD file with the function names across all languages and the corresponding documentation.

Reply:
I **really** want to figure out a way to write API documentation for multiple languages in a centralized way and without replicating the same text multiple times. Does not have to be MD.

Reply:
Swig might help in that regard: https://libpasta.github.io/blog/bindings/ https://github.com/Dushistov/rust_swig